reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -6737,7 +6737,8 @@ public int getPrefetch() {
 	 * @return a new {@link Mono}
 	 */
 	public final Mono<T> publishNext() {
-		return Mono.onAssembly(new MonoProcessor<>(this));
+		//Should add a ConnectableMono to align with #publish()
```

Comment:
Could we add a `TODO` to not forget about this?

Reply:
I think it will be hard to repurpose this method for a `ConnectableMono` if we introduce one. Since we now have `shareNext()`, I'd say that we deprecate this method entirely. Later on, if we introduce `Mono.publish()` returning a `ConnectableMono`, we can advise people to use `Flux.next().publish()` instead. wdyt @smaldini ?